src/kernel/arch/x86/pit.zig

```diff
@@ -0,0 +1,288 @@
+/// then its OUT pit is set high (1). Count down starts then the COUNT is set. The OUT pin remains
+/// high until the counter is reloaded with a new COUNT value or a new control work is written.
+const OCW_MODE_TERMINAL_COUNT: u8           = 0x00; // xxxx000x
+
+/// The counter is programmed to output a pulse every curtain number of clock pulses. The OUT pin
+/// remains high as soon as a control word is written. When COUNT is written, the counter waits
+/// until the rising edge of the GATE pin to start.  One clock pulse after the GATE pin, the OUT
+/// pin will remain low until COUNT reaches 0.
+const OCW_MODE_ONE_SHOT: u8                 = 0x02; // xxxx001x
+
+/// The counter is initiated with a COUNT value. Counting starts next clock pulse. OUT pin remains
+/// high until COUNT reaches 1, then is set low for one clock pulse. Then COUNT is reset back to
+/// initial value and OUT pin is set high again.
+const OCW_MODE_RATE_GENERATOR: u8           = 0x04; // xxxx010x
+
+/// Similar to PIT_OCW_MODE_RATE_GENERATOR, but OUT pin will be high for half the time and low for
+/// half the time. Good for the speaker when setting a tone.
+const OCW_MODE_SQUARE_WAVE_GENERATOR: u8    = 0x06; // xxxx011x
+
+/// The counter is initiated with a COUNT value. Counting starts on next clock pulse. OUT pin remains
+/// high until count is 0. Then OUT pin is low for one clock pulse. Then resets to high again.
+const OCW_MODE_SOFTWARE_TRIGGER: u8         = 0x08; // xxxx100x
+
+/// The counter is initiated with a COUNT value. OUT pin remains high until the rising edge of the
+/// GATE pin. Then the counting begins. When COUNT reaches 0, OUT pin goes low for one clock pulse.
+/// Then COUNT is reset and OUT pin goes high. This cycles for each rising edge of the GATE pin.
+const OCW_MODE_HARDWARE_TRIGGER: u8         = 0x0A; // xxxx101x
+
+/// The counter value is latched into an internal control register at the time of the I/O write
+/// operations.
+const OCW_READ_LOAD_LATCH: u8               = 0x00; // xx00xxxx
+
+/// Read or load the most significant bit only.
+const OCW_READ_LOAD_LSB_ONLY: u8            = 0x10; // xx01xxxx
+
+/// Read or load the least significant bit only.
+const OCW_READ_LOAD_MSB_ONLY: u8            = 0x20; // xx10xxxx
+
+/// Read or load the least significant bit first then the most significant bit.
+const OCW_READ_LOAD_DATA: u8                = 0x30; // xx11xxxx
+
+/// The OCW bits for selecting counter 0. Used for the system clock.
+const OCW_SELECT_COUNTER_0: u8              = 0x00; // 00xxxxxx
+
+/// The OCW bits for selecting counter 1. Was for the memory refreshing.
+const OCW_SELECT_COUNTER_1: u8              = 0x40; // 01xxxxxx
+
+/// The OCW bits for selecting counter 2. Channel for the speaker.
+const OCW_SELECT_COUNTER_2: u8              = 0x80; // 10xxxxxx
+
+// The divisor constant
+// Reviewer: I have found 2 values for this, 1193182 and 1193180, and not sure which one is correct?
```

Comment:
Hmm, since there's very little difference it's probably better to be conservative and choose the lower number.

Reply:
Ill go for the lower one :)